### PR TITLE
[FIX]fix parse k8s version

### LINKED ISSUE #2732: [Bug] The app run status is always STARTING
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### Java Version

1.8

### Scala Version

2.12.x

### StreamPark Version

latest
DIGEST ：cb5b91963239

### Flink Version

1.15.3

### deploy mode

kubernetes-session

### What happened

The app run status is always STARTING when the app runing(JobManager can see the corresponding task also k8s can see the corresponding pod). The kubenetes-session and kubenetes-application both have this problem.

### Error Exception

```log
15:44:00.141 [ForkJoinPool-1-worker-0] WARN org.apache.streampark.flink.kubernetes.watcher.FlinkJobStatusWatcher - Failed to visit remote flink jobs on kubernetes-native-mode cluster, and the retry access logic is performed.
```

### Screenshots

image
image

image
image

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!(您是否要贡献这个PR?)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Jcodelove**
## What changes were proposed in this pull request

Issue Number: close #2732

## Brief change log

fix parse k8s version
- when I use ACK and EKS, the Minor of k8s obtained from the client is not just numbers

### CODE REVIEW

**Comment by senlizishi** [comment]
Hi, @Jcodelove Thank you for your contribution. I have a question here. May I ask if there is any relevant information to prove whether the occurrence of `majorVersion=24+` is due to Alibaba Cloud K8S customization or whether the K8S standard version itself is `24+`

**Comment by Jcodelove** [comment]
> Hi, @Jcodelove Thank you for your contribution. I have a question here. May I ask if there is any relevant information to prove whether the occurrence of `majorVersion=24+` is due to Alibaba Cloud K8S customization or whether the K8S standard version itself is `24+`

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/component-base/version/base.go
According to the above code description, K8S standard version itself is numeric possibly followed by "+".
Maybe I just need to simply deal with the minor field.

**Comment by senlizishi** [comment]
Based on the information you provided, I found that it mentioned '**TODO: Deprecate gitMajor and gitMinor, use only gitVersion**'. I think it may be better for us to use **gitversion** to obtain the version in order to support later versions. What do you think? @Jcodelove

**Comment by Jcodelove** [comment]
fix commit log

**Comment by Jcodelove** [comment]
> Based on the information you provided, I found that it mentioned '**TODO: Deprecate gitMajor and gitMinor, use only gitVersion**'. I think it may be better for us to use **gitversion** to obtain the version in order to support later versions. What do you think? @Jcodelove

I am revising this question, you can see what else is wrong

**Comment by senlizishi** [comment]
LGTM  @Al-assad  cc

**Review by Al-assad — DISMISSED** [review]
LGTM, Thanks for your contribution! 😃

**Comment by Al-assad** [comment]
@Jcodelove Hi Jcodelove, please check the CI errors in this section:
https://github.com/apache/incubator-streampark/actions/runs/4957761895/jobs/8870077942?pr=2741

**Comment by Jcodelove** [comment]
> 

fixed

**Comment by Jcodelove** [comment]
> @Jcodelove Hi Jcodelove, please check the CI errors in this section: https://github.com/apache/incubator-streampark/actions/runs/4957761895/jobs/8870077942?pr=2741

fixed

**Comment by wolfboys** [comment]
hi @Jcodelove Thanks for your contribution,  Can you provide the testcase?

File: streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/ingress/IngressStrategy.scala [hunk 1]
```diff
@@ -26,13 +26,27 @@ import java.io.File
 
 trait IngressStrategy {
 
+  // splits a version string into numeric and "extra" parts
+  private val VERSION_MATCH_RE = "^\\s*v?([0-9]+(?:\\.[0-9]+)*)(.*)*$".r
```
Comment (wolfboys) [inline streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/ingress/IngressStrategy.scala]: We can remove `VERSION_MATCH_RE`

File: streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/ingress/IngressController.scala [hunk 2]
```diff
@@ -30,8 +30,7 @@ object IngressController extends Logger {
   private lazy val ingressStrategy: IngressStrategy = {
     using(new DefaultKubernetesClient()) {
       client =>
-        val versionInfo = client.getVersion
-        val version = s"${versionInfo.getMajor}.${versionInfo.getMinor}".toDouble
+        val version = ingressStrategy.parseSemantic(client.getVersion.getGitVersion)
```
Comment (wolfboys) [inline streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/ingress/IngressController.scala]: val version = VERSION_REGEXP.findFirstIn(client.getVersion.getGitVersion).get.toDouble

File: streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/ingress/IngressStrategy.scala [hunk 3]
```diff
@@ -26,13 +26,27 @@ import java.io.File
...
+  /**
+   * The version information of kubernetes is based on the gitVersion field ParseSemantic parses a
+   * version string that exactly obeys the syntax and semantics of the "Semantic Versioning"
+   * specification (http://semver.org/) (although it ignores leading and trailing whitespace, and
+   * allows the version to be preceded by "v").
+   */
+  def parseSemantic(getGitVersion: String): Double = {
+    val numbers = VERSION_MATCH_RE.findFirstMatchIn(getGitVersion).get.group(1).split('.')
+    s"${numbers(0)}.${numbers(1)}".toDouble
+  }
```
Comment (wolfboys) [inline streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/ingress/IngressStrategy.scala]: We can remove this method

File: streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/ingress/IngressController.scala [hunk 4]
```diff
@@ -30,8 +30,7 @@ object IngressController extends Logger {
   private lazy val ingressStrategy: IngressStrategy = {
```
Comment (wolfboys) [inline streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/ingress/IngressController.scala:32]: defined before line 30:
`private[this] val VERSION_REGEXP = "(\\d+\\.\\d+)".r`

Comment (Jcodelove) [reply streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/ingress/IngressController.scala:32]: > 

completed

**Review by wolfboys — APPROVED** [review]
LGTM

**Pull request merged**